design/ceph/object/store.md

```diff
@@ -73,7 +74,124 @@ If `preservePoolsOnDelete` is set to 'true' the pools used to support the object
+#### Pools shared by multiple CephObjectStore
+
+If user want to use existing pools for metadata and data, the pools must be created before the object store is created. This will be useful if multiple objectstore can share same pools. The detail of pools need to shared in `radosNamespaces` settings in object-store CRD. Now the object stores can consume same pool isolated with different namespaces. User can create via [Pool CRD](/Documentation/CRDs/Block-Storage/ceph-block-pool-crd.md), this is need to present before the object store is created.
+
+```yaml
+  radosNamespaces:
+    metadataPoolName: rgw-meta-pool
+    dataPoolName: rgw-data-pool
+```
+
+To create the pools that will be shared by multiple object stores, create the following CephBlockPool CRs:
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+    name: rgw-meta-pool
+  spec:
+    failureDomain: host
+    replicated:
+      size: 3
+    parameters:
+      pg_num: 8
+    application: rgw
+---
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+ metadata:
+    name: rgw-data-pool
+  spec:
+    failureDomain: osd
+      erasureCoded:
+        dataChunks: 6
+        codingChunks: 2
+    application: rgw
+```
+
+The `.rgw.root` pool must be defined before the object store is created. This pool is used for storing the root info for all object stores.
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+    name: rgw-root
+  spec:
+    name: .rgw.root
+    failureDomain: host
+    replicated:
+      size: 3
+    parameters:
+      pg_num: 8
+    application: rgw
+```
+
+The pools for this configuration will be created as below:
+
+```bash
+# ceph osd pool ls
+.rgw.root
+rgw-meta-pool
+rgw-data-pool
+```
+
+And the pool configuration in zone is as below:
+
+```json
+# radosgw-admin zone get --rgw-zone=my-store
+{
+    "id": "2220eb5f-2751-4a51-9c7d-da4ce1b0e4e1",
+    "name": "my-store",
+    "domain_root": "rgw-meta-pool:my-store.root",
+    "control_pool": "rgw-meta-pool:my-store.control",
+    "gc_pool": "rgw-meta-pool:my-store.gc",
+    "lc_pool": "rgw-meta-pool:my-store.lc",
+    "log_pool": "rgw-meta-pool:my-store",
+    "intent_log_pool": "rgw-meta-pool:my-store.intent",
+    "usage_log_pool": "rgw-meta-pool:my-store.usage",
+    "roles_pool": "rgw-meta-pool:my-store.roles",
+    "reshard_pool": "rgw-meta-pool:my-store.reshard",
+    "user_keys_pool": "rgw-meta-pool:my-store.users.keys",
+    "user_email_pool": "rgw-meta-pool:my-store.users.email",
+    "user_swift_pool": "rgw-meta-pool:my-store.users.swift",
+    "user_uid_pool": "rgw-meta-pool:my-store.users.uid",
+    "otp_pool": "rgw-meta-pool:my-store.otp",
+    "system_key": {
+        "access_key": "",
+        "secret_key": ""
+    },
+    "placement_pools": [
+        {
+            "key": "default-placement",
+            "val": {
+                "index_pool": "rgw-data-pool:my-store.buckets.index",
```

Comment:
Index goes with data and not metadata?

Rook currently treats the index as a meta pool: https://github.com/rook/rook/blob/719a8cdb50712115eece923db3bbe724ffbe7c8d/pkg/operator/ceph/object/objectstore.go#L59-L67

Is this a bug in Rook's current implementation? Is this something that some users might want more control over? Might some users want this to be a data pool?

Reply:
@BlaineEXE From the [docs](https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/4/html/object_gateway_for_production_guide/assembly-developing-storage-strategies-rgw-prod#creating-service-pools-rgw-prod), data pools are `rgw.buckets.data` and `rgw.buckets.non-ec`. The `rgw.buckets.non-ec` store objects related to multipart upload(not the head object) if the data pool is configured to `ec`

Reply:
Thats bug in Rook atm

Reply:
Where does that doc indicate the `buckets.index` pool is a data pool? An index should generally be replicated for higher performance, and not erasure-coded. Only the data pools should be erasure-coded. So I believe this should be in the `rgw-metadata-pool`.
```suggestion
                "index_pool": "rgw-metadata-pool:my-store.buckets.index",
```

Reply:
@cbodley please advise

Reply:
> An index should generally be replicated for higher performance, and not erasure-coded. Only the data pools should be erasure-coded. So I believe this should be in the rgw-metadata-pool.

agree with @travisn here. the bucket index uses omap exclusively so can't run on an erasure-coded pool

is rook's distinction between 'metadata' and 'data' pools strictly about erasure-coding, or are there other considerations too?

Reply:
@travisn : `bucket.index` is not a data pool, I totally agree with that. I was only referring to  the `rgw.buckets.non-ec` is datapool but cannot be configured to EC from the doc

Reply:
> is rook's distinction between 'metadata' and 'data' pools strictly about erasure-coding, or are there other considerations too?

Rook's distinction was also to create the metadata pools with lower PG count.

Reply:
@cbodley we have been specfically discussing `bucket.index`, but there was another pool in question: `rgw.buckets.non-ec`. By the "non-ec" name, I assume that the same thing holds true here. Is that a safe assumption?

Reply:
the non-ec pool also relies on omap to track the parts of multipart uploads, so can't be erasure-coded. i don't think we store object data in it, so it's usage is similar to the index pool